README.md

```diff
@@ -0,0 +1,152 @@
+# bazel_rules_detekt
+
+Integration of [Detekt](https://github.com/arturbosch/detekt) — Kotlin static analysis tool for [Bazel build system](https://bazel.build).
+
+***Table of Contents***
+
+- [Overview](#overview)
+    - [Features](#features)
+    - [TODOs](#todos)
+- [Usage](#usage)
+    - [`WORKSPACE` Configuration](#workspace-configuration)
+    - [`BUILD` Configuration](#build-configuration)
+    - [Executing](#executing)
+    - [Supported Attributes](#supported-attributes)
+        - [`srcs`](#srcs)
+        - [`config`](#config)
+        - [`parallel`](#parallel)
+        - [`xml_report`](#xml_report)
+        - [`html_report`](#html_report)
+
+## Overview
+
+### Features 
+
+- User-Provided Detekt Config Files
+- TXT, XML and HTML Reporting
+- Parallel Detekt "Compilation" 
+
+### TODOs
+
+- Baseline Files (blocked by Detekt outputting and reading absolute paths in baselines, see [#3](https://github.com/buildfoundation/bazel_rules_detekt/issues/3))
+- Persistent Worker Mode
+- User-Provided Detekt CLI Jar (blocked by Persistent Worker Mode due to potential changes in classloading/etc, see [#14](https://github.com/buildfoundation/bazel_rules_detekt/issues/14))
+
+
+## Usage
+
+### `WORKSPACE` Configuration
+
+First of all you need to declare `rules_detekt` in `WORKSPACE`:
+
+(for version and sha256 see [GitHub Releases page](https://github.com/buildfoundation/bazel_rules_detekt/releases))
+
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+rules_detekt_version = "see-github-releases-page"
+rules_detekt_sha256 = "see-github-releases-page"
+
+http_archive(
+    name = "rules_detekt",
+    urls = ["https://github.com/buildfoundation/bazel_rules_detekt/archive/%s.zip" % rules_detekt_version],
+    type = "zip",
+    strip_prefix = "rules_detekt-%s" % rules_detekt_version,
+    sha256 = rules_detekt_sha256,
+)
+```
+
+### `BUILD` Configuration
+
+Once declared in `WORSKPACE`, `rules_detekt` can be `load`ed and applied in a `BUILD` file:
+
+```python
+load("@rules_detekt//detekt:detekt.bzl", "detekt")
+
+detekt(
+    name = "my_detekt",
```

Comment:
`main`? Since reports are auto-suffixed with `detekt_report` it will be weird to see `my_detekt_detekt_report`.

Reply:
Names are such a custom thing in Bazel, different teams have different conventions and name has to be unique within package (read module), it's very likely `main` will be occupied by actual `kt_jvm_library` for example

So whatever we set here is pretty much irrelevant and is just an example :)